internal/store/kubernetesdiscoverys/reducers.go

```diff
@@ -88,21 +89,28 @@ func RefreshKubernetesResource(state *store.EngineState, name string) {
 			}
 
 			krs.FilteredPods = r.FilteredPods
+			krs.Conditions = r.ApplyStatus.Conditions
 
 			isReadyOrSucceeded := false
-			for _, pod := range r.FilteredPods {
-				if krs.PodReadinessMode == model.PodReadinessSucceeded {
-					// for jobs, we don't care about whether it's ready, only whether it's succeeded
-					isReadyOrSucceeded = pod.Phase == string(v1.PodSucceeded)
-				} else {
-					isReadyOrSucceeded = len(pod.Containers) != 0 && store.AllPodContainersReady(pod)
-				}
-				if isReadyOrSucceeded {
-					// NOTE(nick): It doesn't seem right to update this timestamp everytime
-					// we get a new event, but it's what the old code did.
-					krs.LastReadyOrSucceededTime = time.Now()
+			if len(r.FilteredPods) != 0 {
+				for _, pod := range r.FilteredPods {
+					if krs.PodReadinessMode == model.PodReadinessSucceeded {
+						// for jobs, we don't care about whether it's ready, only whether it's succeeded
+						isReadyOrSucceeded = pod.Phase == string(v1.PodSucceeded)
```

Comment:
unrelated to this pr, but i'm pretty sure this logic is wrong? this will use the readiness state of the last pod (and the order of pods is non-deterministic).

maybe break this out so that it's only ready if all pods are ready? (The FilteredPods will take care of filtering old pods)